cmd/http-check/main.go

```diff
@@ -22,43 +26,94 @@ func init() {
 		log.Fatalln("Empty URL. Please update your CHECK_URL environment variable.")
 	}
 
+	// Check that the COUNT environment variable is valid.
+	if len(count) == 0 {
+		log.Fatalln("Empty count value. Please update your COUNT environemnt variable.")
+	}
+
+	// Check that the SECONDS environment variable is valid.
+	if len(seconds) == 0 {
+		log.Fatalln("Empty seconds value. Please update your SECONDS environemnt variable.")
+	}
+
 	// If the URL does not begin with HTTP, exit.
 	if !strings.HasPrefix(checkURL, "http") {
 		log.Fatalln("Given URL does not declare a supported protocol. (http | https)")
 	}
 }
 
 func main() {
-	log.Infoln("Beginning check.")
 
-	// Make a GET request.
-	r, err := http.Get(checkURL)
+	countInt, err := strconv.Atoi(count)
 	if err != nil {
-		reportErr := fmt.Errorf("error occurred performing a " + http.MethodGet + " to " + checkURL + ": " + err.Error())
-		log.Errorln(reportErr.Error())
-		reportFailureErr := kh.ReportFailure([]string{reportErr.Error()})
-		if reportFailureErr != nil {
-			log.Fatalln("error when reporting to kuberhealthy:", reportFailureErr.Error())
-		}
-		os.Exit(0)
+		log.Fatalln(err)
 	}
 
-	// Check if the response status code is a 200.
-	if r.StatusCode == http.StatusOK {
+	secondInt, err := strconv.Atoi(seconds)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	passingFloat, err := strconv.ParseFloat(passing, 64)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	// sets the passing score to compare it against checks that have been ran
+	floatCount := float64(countInt)
+	passingScore := floatCount * passingFloat
+	passInt := int(passingScore)
+	log.Infoln("Looking for at least", passInt, "of", countInt, "checks to pass")
+
+	log.Infoln("Beginning check.")
+	checksRan := 0
+	checksPassed := 0
+	checksFailed := 0
+
+	ticker := time.NewTicker(time.Duration(secondInt))
+	defer ticker.Stop()
+
+	// This for loop makes a http GET request to a known internet address, address can be changed in deployment spec yaml
+	// and returns a http status every second.
+	for checksRan < countInt {
+		<-ticker.C
+		r, err := http.Get(checkURL)
+		checksRan++
+
+		if err != nil {
+			checksFailed++
+			log.Infoln("Failed to reach URL: ", checkURL)
+			continue
+		}
+
+		if r.StatusCode != http.StatusOK {
+			log.Infoln("Got a", r.StatusCode, "with a", http.MethodGet, "to", checkURL)
```

Comment:
use `log.Errorln` here or even better yet, produce some summary of bad status codes and how many times they were seen, then report that to Kuberhealthy when failure occurs.

Reply:
I changed this to an `log.Errorln`